knowledge_repo/app/routes/web_editor.py

```diff
@@ -88,228 +66,176 @@ def gitless_drafts():
     current_repo.publish(path)
-    return ""
 
 
 @blueprint.route('/unpublish_post', methods=['GET', 'POST'])
 @PageView.logged
 def unpublish_post():
-    """ Unpublish the post, going back to the in_review status """
-    post_id = request.args.get('post_id', None)
-    post = (db_session.query(Post)
-                      .filter(Post.id == post_id)
-                      .first())
-    post.status = current_repo.PostStatus.UNPUBLISHED
-    db_session.commit()
-
-    path = post.path
+    """ Unpublish the post """
+    path = request.args.get('path', None)
+    if not path not in current_repo:
+        return json.dumps({'msg': "Unable to retrieve post with path = {}!".format(path), 'success': False})
     current_repo.unpublish(path)
-    return ""
 
 
 @blueprint.route('/author_publish', methods=['POST'])
 @PageView.logged
 def change_author_publish():
-    """ Allow the author to publish their own webpost if the
-        editor allows them to
-    """
-    post_id = request.args.get('post_id', None)
-    post = (db_session.query(Post)
-                      .filter(Post.id == post_id)
-                      .first())
-    post.status = current_repo.PostStatus.UNPUBLISHED
-    db_session.commit()
-
-    path = post.path
+    """ Accept the post """
+    path = request.args.get('path', None)
+    if path not in current_repo:
+        return json.dumps({'msg': "Unable to retrieve post with path = {}!".format(path), 'success': False})
     current_repo.accept(path)
-    return ""
 
 
-@blueprint.route('/save_post', methods=['GET', 'POST'])
+@blueprint.route('/delete_post', methods=['GET'])
 @PageView.logged
-def save_post():
-    """ Save the gitless post """
-    post_id = request.args.get('post_id', None)
-    data = request.get_json()
-    post = (db_session.query(Post)
-            .filter(Post.id == post_id)
-            .first())
-    new_post = False
-    if not post:
-        new_post = True
-        path = "{}/{}.kp".format(data['project'],
-                                 data['title'].encode('utf8').lower().replace(' ', '_'))
-        if current_app.config.get('WEB_EDITOR_PREFIXES', None):
-            # TODO: Include dropdown on webeditor to have user specify repo
-            path = "{}/{}".format(current_app.config['WEB_EDITOR_PREFIXES'][0], path)
-
-        post = (db_session.query(Post)
-                          .filter(Post.path == path)
-                          .first())
-        if post:
-            error_msg = "Post with project {} and title {} already exists!".format(data['project'], data['title'])
-
-            json_str = json.dumps({'msg': error_msg, 'success': False})
-            return json_str
-        else:
-            post = Post()
-            post.path = path
-    else:
-        path = post.path
-
-    # create the knowledge post
-    kp = KnowledgePost(path=path)
+def delete_post():
+    """ Delete a post """
```

Comment:
so you could have multiple routes point to the same function - something like 
```
@blueprint.route('/delete_post')
@blueprint.route('/accept_post')
def do_the_function()
    return "ok"
```
The reason I suggested this was because for most of the functions you seem to be getting the path, checking to see if the path is in the current_repo, and then returning something based on that.